android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/NativeRadioButtonFactory.java

```diff
@@ -598,7 +598,20 @@ public void run() {
             radioButton.setTextSize(FormUtils.getValueFromSpOrDpOrPx(optionTextSize, this.context));
             radioButton.setText(optionText);
             radioButton.setEnabled(!readOnly);
-            radioButton.setOnCheckedChangeListener(listener);
+
+            // Fix listener bug on older Android versions
+            if (!TextUtils.isEmpty(jsonObject.optString(JsonFormConstants.VALUE)) &&
+                    jsonObject.optString(JsonFormConstants.VALUE).equals(item.getString(JsonFormConstants.KEY))) {
+                ((Activity) context).runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        radioButton.setChecked(true);
+                        radioButton.setOnCheckedChangeListener(listener);
+                    }
+                });
+            } else {
+                radioButton.setOnCheckedChangeListener(listener);
+            }
```

Comment:
What do you mean by older versions? What version is this supposed to handle? Do you mind adding a check for the version. If left as is it will run for all versions

Reply:
Basically on Android versions 7 and below, the `runOnUiThread` was being called with a delay, triggering `onCheckedChanged` for the button when the value was set programmatically. So I just fixed the code flow for the button to always register the listener after setting the value programatically. This will work for all Android versions. Let me update the misleading comment.